example/README.md

```diff
@@ -1,23 +1,57 @@
-An simple example of a Dart package using [json_serializable].
+To use [package:json_serializable][json_serializable] in your package, add these
+dependencies to your `pubspec.yaml`.
 
-* `lib/example.dart`: A library configured with annotations – 
-  `JsonSerializable` and `JsonLiteral` – to enable code generation.
+```yaml
+dependencies:
+  json_annotation: ^0.2.1
 
-  * Note: the annotations are defined in `package:json_annotation`.
-    This is the only package required in the `dependencies` section of your
-    `pubspec.yaml`.
+dev_dependencies:
+  build_runner: ^0.6.1
+  json_serializable: ^0.2.5
+  source_gen: ^0.7.2+1
+```
 
-* `tool/`: Contains the code run during development to create and update
-  generated code.
+Next, create the scripts to run code generation. The [`tool`][tool] directory
+contains the conventional layout of these scripts.
 
-  * `build_actions.dart`: A convention when using `package:build` to 
-    have one location to define the actions to be run by `build.dart` and 
-    `watch.dart`. See the comments in the source file for more information.
+* [`build_actions.dart`][tool]: A convention when using `package:build` to 
+  have one location to define the actions to be run by [`build.dart`][build] and 
+  [`watch.dart`][watch]. See the comments in the source file for more
+  information.
 
-  * `build.dart`: Runs one build using the actions defined in
-    `build_actions.dart`.
+* [`build.dart`][build]: Runs one build using the actions defined in
+  [`build_actions.dart`][build_actions].
 
-  * `watch.dart`: Starts a watcher that (re)runs the actions defined in
-    `build_actions.dart` when files change.
+* [`watch.dart`][watch]: Starts a watcher that (re)runs the actions defined in
+  [`build_actions.dart`][build_actions] when files change.
 
+Finally, annotate your code with classes defined in
+[package:json_annotation][json_annotation].
+
+* See [`lib/example.dart`][example] for an example of a file using these
+  annotations.
+
+* See [`lib/example.g.dart`][example_g] for the generated file.
+
+Run the build script to update the generated code after you modify of of the
+source files.
+
+```console
+> dart bin/build.dart
```

Comment:
@sethladd – how does a flutter dev invoke a dart script?

Reply:
today, it's a bit gnarly (@mit-mit is looking to improve this), but it's:

`flutter packages pub run package:script`

See https://github.com/flutter/flutter/issues/13360 which asks to make this better.

Reply:
Sadly, this won't work. It's not in the bin directory, which is the only place `pub run` will run scripts.